pkg/machinery/go.mod

```diff
@@ -9,7 +9,8 @@ replace gopkg.in/yaml.v3 => github.com/unix4ever/yaml v0.0.0-20220527175918-f17b
 require (
 	github.com/blang/semver/v4 v4.0.0
 	github.com/containerd/go-cni v1.1.9
-	github.com/cosi-project/runtime v0.3.20
+	github.com/coredns/coredns v1.11.1
```

Comment:
that looks bad - we should never import it from machinery

Reply:
so if the logic of the PR enforces this resource to reference a "handle" from coredns package, we should rather not to it at all :(

Reply:
So I dropped the concrete type in "machinery" for the interface. It requires a small type-asssertion in the controllers, but otherwise it's pretty much the same. Except it machinery no longer needs "coredns".

We could use generics there, but in the absence of generic aliases the usage will look ugly.